Updated README on debugging web and python (#2544)
infiniflow/ragflow: README.md

To update the default HTTP serving port (80), go to [docker-compose.yml](./docker/docker-compose.yml) and change `80:80` to `<YOUR_SERVING_PORT>:80`.

> Updates to above configurations require a reboot of containers to take effect:

Reviewer: ```suggestion
Updates to the above configurations require a reboot of all containers to take effect:
```

   $ cd ragflow/
   ```

> 2. Install all python dependencies to a newly created virtual environment `.venv`:

Reviewer: ```suggestion
2. Install all python dependencies in a newly created virtual environment named `.venv`:
```

   ```bash
   $ curl -sSL https://install.python-poetry.org | python3 -
   $ $HOME/.local/bin/poetry install --sync --no-root
   ```

3. Copy the entry script and configure environment variables:
   
   ```bash
   # Adjust configurations according to your actual situation (the following two export commands are newly added):
   # - Comment out `LD_LIBRARY_PATH`, if it is configured.
   # - Optional: Add Hugging Face mirror.
   source ~/.venv/bin/activate
   export PYTHONPATH=$(pwd)
   export HF_ENDPOINT=https://hf-mirror.com
   ```

4. Launch the third-party services (MinIO, Elasticsearch, Redis, and MySQL):

   ```bash
   $ docker compose -f docker/docker-compose-base.yml up -d
   ```

> 5. Add following line to `/etc/hosts` so that hosts in `conf/service_conf.yaml` are all resolved to `127.0.0.1`:

Reviewer: ```suggestion
5. Add the following line to `/etc/hosts` to resolve all hosts in `conf/service_conf.yaml` to `127.0.0.1`:
```

   ```
   127.0.0.1       localhost localhost.localdomain es01 mysql minio redis
   ```

6. Launch the RAGFlow backend service:

>    Comment `nginx` line in `docker/entrypoint.sh` and run:

Reviewer: ```suggestion
   Comment out the `nginx` line in `docker/entrypoint.sh` and run the script:
```

   ```bash
   $ bash docker/entrypoint.sh
   ```

7. Launch the frontend service:

   ```bash
   $ cd web
   $ npm install --force
   $ vim .umirc.ts
   # Update proxy.target to http://127.0.0.1:9380
   $ npm run dev 
   ```

> 8. In web browser, enter `http://127.0.0.1/`.

Reviewer: ```suggestion
8. In your web browser, enter `http://127.0.0.1/`.
```

   $ cd ragflow/
   ```

2. Install all python dependencies in a newly created virtual environment named `.venv`:

   ```bash
   $ curl -sSL https://install.python-poetry.org | python3 -
   $ $HOME/.local/bin/poetry install --sync --no-root
   ```

3. Copy the entry script and configure environment variables:
   
   ```bash
   # Adjust configurations according to your actual situation (the following two export commands are newly added):
   # - Comment out `LD_LIBRARY_PATH`, if it is configured.
   # - Optional: Add Hugging Face mirror.
   source ~/.venv/bin/activate
   export PYTHONPATH=$(pwd)
   export HF_ENDPOINT=https://hf-mirror.com
   ```

4. Launch the third-party services (MinIO, Elasticsearch, Redis, and MySQL):

   ```bash
   $ docker compose -f docker/docker-compose-base.yml up -d
   ```

5. Adjust configurations
   Add the following line to `/etc/hosts` to resolve all hosts in `docker/service_conf.yaml` to `127.0.0.1`:

   ```
   127.0.0.1       es01 mysql minio redis
   ```

>    Edit `docker/service_conf.yaml` to change mysql port to `5455` and es port to `1200` according to `docker/.env`.

Reviewer: ```suggestion
   Edit `docker/service_conf.yaml` to change mysql port to `5455` and es port to `1200`, as specified in `docker/.env`.
```

   $ cd ragflow/
   ```

2. Install all python dependencies in a newly created virtual environment named `.venv`:

   ```bash
   $ curl -sSL https://install.python-poetry.org | python3 -
   $ $HOME/.local/bin/poetry install --sync --no-root
   ```

3. Copy the entry script and configure environment variables:
   
   ```bash
   # Adjust configurations according to your actual situation (the following two export commands are newly added):
   # - Comment out `LD_LIBRARY_PATH`, if it is configured.
   # - Optional: Add Hugging Face mirror.
   source ~/.venv/bin/activate
   export PYTHONPATH=$(pwd)
   export HF_ENDPOINT=https://hf-mirror.com
   ```

4. Launch the third-party services (MinIO, Elasticsearch, Redis, and MySQL):

   ```bash
   $ docker compose -f docker/docker-compose-base.yml up -d
   ```

5. Adjust configurations
   Add the following line to `/etc/hosts` to resolve all hosts in `docker/service_conf.yaml` to `127.0.0.1`:

   ```
   127.0.0.1       es01 mysql minio redis
   ```

   Apply following patch to change services ports in `docker/service_conf.yaml` according to `docker/.env`:
   ```

> diff --git a/docker/service_conf.yaml b/docker/service_conf.yaml

Reviewer 2: ????
